microphone/drv_audio_pdm.h

```diff
@@ -8,7 +8,7 @@
 
 #define PDM_BUF_NUM 	2
 #define PDM_BUF_SIZE 	2048
-#define DECIMATION		32
+#define DECIMATION		16 // 20KHz/16 = 1.250 KHz sample rate for LO position 
```

Comment:
Add assert that buffer needs to be a size that is divisible by the low sample rate

Reply:
I think, we also need to change the `test.py`. The `PDM_CLK_DIV` frequency is changed https://github.com/TUDelft-SPC-Lab/midge-code/blob/master/BadgeFramework/test.py#L22 and https://github.com/TUDelft-SPC-Lab/midge-code/blob/master/BadgeFramework/test.py#L62 . That means the `assert` would fail for them, wouldn't it?

Reply:
added both the static assert and fixed the code that was decoding the frequency value. Also updated the `test.py` file to make sure we check for the "overclocked" PDM freq